File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.cpp
class Solution {
public:
     string longestPalindrome(string s) {
    string res = "";
    int resLen = 0;
    for (int i = 0; i < s.size(); i++) {
      for (int j = i; j < s.size(); j++) {
        int l = i;
        int r = j;
        while (l < r && s[l] == s[r]) {
          r--;
          l++;
        }
        if (l >= r && resLen < (j - i + 1)) {
          res = s.substr(i, j - i + 1);
          resLen = j - i + 1;
        }
      }
    }
    return res;
  }

};

File: Data Structures & Algorithms/daily-temperatures/submission-1.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        stack = []

        for i in range(n):
            while stack and temperatures[i] > stack[-1][0]:
                t, idx = stack.pop()
                res[idx] = (i - idx)
            stack.append([temperatures[i], i])
        return res
            